spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/SparkMLTrainer.java

```diff
@@ -0,0 +1,152 @@
+
+    protected MLTrainerConfig(String fileSetName, String path, @Nullable String featureFieldsToInclude,
+                           @Nullable String featureFieldsToExclude, String labelField) {
+      this.fileSetName = fileSetName;
+      this.path = path;
+      this.featureFieldsToInclude = featureFieldsToInclude;
+      this.featureFieldsToExclude = featureFieldsToExclude;
+      this.labelField = labelField;
+    }
+
+    protected void validate(Schema inputSchema) {
+      SparkUtils.validateConfigParameters(inputSchema, featureFieldsToInclude, featureFieldsToExclude, labelField,
+                                          null);
+      SparkUtils.validateLabelFieldForTrainer(inputSchema, labelField);
+    }
+  }
+
+  public SparkMLTrainer(MLTrainerConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    pipelineConfigurer.createDataset(config.fileSetName, FileSet.class);
+    Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validate(inputSchema);
+  }
+
+  @Override
+  public void prepareRun(SparkPluginContext context) throws Exception { }
+
+  protected Map<String, Integer> getFeatures(Schema inputSchema, @Nullable String featuresToInclude,
+                                             @Nullable String featuresToExclude, String labelField) {
+    return SparkUtils.getFeatureList(inputSchema, featuresToInclude, featuresToExclude, labelField);
+  }
+
+  public static JavaRDD<LabeledPoint> getTrainingData(final Map<String, Integer> fields,
+                                                      JavaRDD<StructuredRecord> input, final String labelField) {
+    JavaRDD<LabeledPoint> trainingData = input.map(new Function<StructuredRecord, LabeledPoint>() {
+      @Override
+      public LabeledPoint call(StructuredRecord record) throws Exception {
+        List<Double> featureList = new ArrayList<>();
+        List<Integer> featureIndex = new ArrayList<>();
+        int counter = 0;
+        for (String field : fields.keySet()) {
+          if (record.get(field) != null) {
+            featureList.add(((Number) record.get(field)).doubleValue());
+            featureIndex.add(counter);
+          }
+          counter++;
+        }
+        Double prediction = record.get(labelField);
+        if (prediction == null) {
+          throw new IllegalArgumentException(String.format("Value of Label field %s value must not be null.",
+                                                           labelField));
+        }
+        return new LabeledPoint(prediction, Vectors.sparse(counter, Ints.toArray(featureIndex),
+                                                           Doubles.toArray(featureList)));
+      }
+    });
+    return trainingData;
+  }
+
+  public SparkContext getSparkContext(SparkExecutionPluginContext context) {
```

Comment:
this can be removed as it doesn't provide much utility. Users can just do context.getSparkContext().sc()